lib/media/drm_engine.js

```diff
@@ -434,10 +434,10 @@ shaka.media.DrmEngine = class {
     this.destroyer_.ensureNotDestroyed();
 
     this.createOrLoad();
-    if (!this.currentDrmInfo_.initData.length &&
-        !this.offlineSessionIds_.length) {
-      // Explicit init data for any one stream or an offline session is
-      // sufficient to suppress 'encrypted' events for all streams.
+
+    // Explicit init data an offline session is
+    // sufficient to suppress 'encrypted' events for all streams.
+    if (!this.offlineSessionIds_.length) {
```

Comment:
ℹ️ We need to have the encrypted event to get the proper initData, since we can't build from manifest data (at least for HLS implementation)